syntax = "proto3";

package server;

service DioDBServer {
  rpc GetDBInfo() returns (DBInfo) {}
  rpc HolyDiver() returns (HolyDiverLyrics) {}
}

message DBInfo {
  // The current version of DioDB running.
  string db_version = 1;

  // The number of keys stored in the database.
  int64 num_keys = 2;

  // The number of bytes stored in the database.
  int64 num_bytes = 3;
}

message HolyDiverLyrics {
  string lyrics = 1;
}
